compiler/src/iree/compiler/Codegen/Common/FlattenMemRefSubspanPass.cpp

```diff
@@ -127,15 +127,17 @@ static Value createTotalElementCountValue(ShapedType type,
   }
 
   int dynamicDimIndex = 0;
-  SmallVector<Value> dims;
+  SmallVector<OpFoldResult> dims;
   auto shape = type.getShape();
   AffineExpr sizeExpr = getAffineConstantExpr(1, context);
   for (int i = 0; i < shape.size(); ++i) {
     sizeExpr = sizeExpr * getAffineSymbolExpr(i, context);
     if (ShapedType::isDynamic(shape[i])) {
       dims.push_back(dynamicDims[dynamicDimIndex++]);
     } else {
-      dims.push_back(builder.create<arith::ConstantIndexOp>(loc, shape[i]));
+      // dims.push_back(builder.getIndexArrayAttr(shape[i]));
```

Comment:
This is not what is needed here. Do you need me to take a look.

Reply:
Sorry, looks like my editor goofed and didn't save my latest changes. 

Removed the commented out line. 

Your suggestion you sent me earlier didn't actually work. It led to this test failures in multiple tests: https://github.com/openxla/iree/actions/runs/5347470211/jobs/9696000319?pr=14188

Using `getValue()` instead resolved the issue.

Reply:
I see those errors now. I am taking a look.

Reply:
For reference, this is the change that started this: https://github.com/iree-org/iree-llvm-fork/commit/efc290ce9c2b85bbd66ebc9ea43986fe89cbff15